### PR TITLE
feat(protocol): Change lock_reason to a held_locks dictionary

### PR DESCRIPTION
**Pull request by romtsn**
A thread can hold multiple locks, so this PR changes the recently introduced `lock_reason` attribute to a dictionary/map, where the key is the lock object address and the value is the existing `LockReason` struct. I was also thinking to make it just an Array, but I think it'd be easier and faster to look up by a lock address when we analyze.

Additionally, I haven't found any docs about how to update snapshot files, so I've added a section to readme about that, lemme know if that makes sense.

https://cs.android.com/android/platform/superproject/+/master:development/tools/bugreport/src/com/android/bugreport/stacks/ThreadSnapshot.java;l=53

### CODE REVIEW

**Comment by romtsn** [comment]
> Additionally, I haven't found any docs about how to update snapshot files, so I've added a section to readme about that, lemme know if that makes sense.

Ah, ok, it's here https://develop.sentry.dev/rust/#development-environment, should I remove it from the readme then?

**Review by olksdr — APPROVED** [review]
lgtm!

**Review by shruthilayaj — APPROVED** [review]

File: relay-general/src/protocol/thread.rs [hunk 1]
```diff
@@ -233,9 +233,10 @@ pub struct Thread {
     #[metastructure(skip_serialization = "empty")]
     pub state: Annotated<String>,
 
-    /// Represents an instance of a held lock (java monitor object) in a thread.
-    #[metastructure(skip_serialization = "empty")]
-    pub lock_reason: Annotated<LockReason>,
+    /// Represents a collection of locks (java monitor objects) held by a thread.
+    ///
+    /// A map of lock object addresses and their respective lock reason/details.
+    pub held_locks: Annotated<Object<LockReason>>,
```
Comment (iker-barriocanal) [inline relay-general/src/protocol/thread.rs:239]: Consider any backward-compatibility if there already are SDKs released with `lock_reason` in the protocol.

**Review by iker-barriocanal — APPROVED** [review]

File: relay-general/src/protocol/thread.rs (hunk 1 above)
Comment (romtsn) [reply relay-general/src/protocol/thread.rs:239]: no, not yet, so we're good here. but thanks for the reminder!

**Pull request merged**